### PR TITLE
chore(components): [date-picker] export instasnce type

### LINKED ISSUE #13586: [Component] [date-picker] 无法通过 InstanceType 获取 ElDatePicker 实例
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.3.7`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36 Edg/113.0.1774.35`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHJlYWN0aXZlIH0gZnJvbSAndnVlJztcbmltcG9ydCB7IEVsRGF0ZVBpY2tlciB9IGZyb20gJ2VsZW1lbnQtcGx1cyc7XG5cbmNvbnN0IG1vbnRoUmFuZ2VSZWYgPSByZWY8SW5zdGFuY2VUeXBlPHR5cGVvZiBFbERhdGVQaWNrZXI+PigpO1xuXG5jb25zdCBwaWNrZXJGb3JtID0gcmVhY3RpdmUoe1xuICBtb250aFJhbmdlOiAnJyxcbiAgZGF0ZXRpbWVSYW5nZTogJycsXG59KTtcblxuc2V0VGltZW91dCgoKSA9PiB7XG4gIG1vbnRoUmFuZ2VSZWYudmFsdWUhLmhhbmRsZU9wZW4oKTtcbn0sIDEwMDApO1xuXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuPGVsLWRhdGUtcGlja2VyIHJlZj1cIm1vbnRoUmFuZ2VSZWZcIiB2LW1vZGVsPVwicGlja2VyRm9ybS5tb250aFJhbmdlXCIgdHlwZT1cIm1vbnRocmFuZ2VcIj48L2VsLWRhdGUtcGlja2VyPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=)

### Steps to reproduce

复制代码到 VSCode 中

### What is Expected?

在VSCode中不报错

### What is actually happening?

在VSCode中报错

### Additional comments

_(empty)_
image

### LINKED ISSUE #13589: [Component] [input] el-input password 不是 type 的有效值。预期值: text或textarea
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.7`
- Browser / OS: `Edge 114.0.1823.82/Windows 11`
- Build Tool: `Vite`

### Related Component

- `el-input`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3QgbXNnID0gcmVmKCdIZWxsbyBXb3JsZCEnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGgxPnt7IG1zZyB9fTwvaDE+XG4gIDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgdHlwZT1cInBhc3N3b3JkXCIvPlxuXG4gIDxwPlxuICAgIDxlbC1pY29uIGNvbG9yPVwidmFyKC0tZWwtY29sb3ItcHJpbWFyeSlcIj48RWxlbWVudFBsdXMgLz48L2VsLWljb24+XG4gICAgRWxlbWVudCBQbHVzIHt7IGVwVmVyc2lvbiB9fSArIFZ1ZSB7eyB2dWVWZXJzaW9uIH19XG4gIDwvcD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

```vue
<template>
  <el-input v-model="msg" type="password"/>
</template>
```

### What is Expected?

预期值：text、textarea、...

### What is actually happening?

password 不是 type 的有效值。预期值: text或textarea

### Additional comments

idea对type="password"标黄，警告：password 不是 type 的有效值。预期值: text或textarea

### PR DESCRIPTION
**Pull request by makedopamine**
closes #13586

### CODE REVIEW

File: packages/components/date-picker/index.ts [hunk 1]
```diff
@@ -13,3 +15,8 @@ export default _DatePicker
 export const ElDatePicker = _DatePicker
 export * from './src/constants'
 export * from './src/props/date-picker'
+
+export type DatePickerInstance = ComponentPublicInstance<
+  DatePickerProps,
+  DatePickerExpose
+>
```
Comment (btea) [inline packages/components/date-picker/index.ts:22]: This can be put into an `instance.ts` file separately.

Comment (makedopamine) [reply packages/components/date-picker/index.ts]: Updated, PTAL.

**Review by btea — APPROVED** [review]
LGTM

File: packages/components/date-picker/src/instance.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import type { ComponentPublicInstance } from 'vue'
+import type { DatePickerProps } from './props/date-picker'
+
+export type DatePickerExpose = {
+  focus: (focusStartInput: boolean) => void
+  handleOpen: () => void
+  handleClose: () => void
+}
+export type DatePickerInstance = ComponentPublicInstance<
```
Comment (ryuhangyeong) [inline packages/components/date-picker/src/instance.ts:9]: @btea 

It's a different way from instances of other existing components, does it matter?

## Example

```
// /collapse/src/instance.ts
import type Collapse from './collapse.vue'
import type CollapseItem from './collapse-item.vue'

export type CollapseInstance = InstanceType<typeof Collapse>
export type CollapseItemInstance = InstanceType<typeof CollapseItem>
```

**Review by tolking — APPROVED** [review]

**Pull request merged**